### ISSUE TITLE
rotate bug, possibility to turn off autoPanIfNeeded

### ISSUE THREAD

**Issue by farmasek** [chrvadala/react-svg-pan-zoom#22]
hi, i am using matrix transformation to rotate svg - calculate custom a,b,c,d,e,f from original to rotate and setValue it to Viewer. 

When it's rotated, and you try to pan, the svg jumps to center, because of autoPanIfNeeded method, and you cant move outside of viewbox on some sides. 

thats how i rotate matrix:

```
const multiply = (A, B) => ({
  a: A.a*B.a+A.c*B.b,
  b: A.b*B.a+A.d*B.b,
  c: A.a*B.c+A.c*B.d,
  d: A.b*B.c+A.d*B.d,
  e: A.a*B.e+A.c*B.f+A.e,
  f: A.b*B.e+A.d*B.f+A.f,
});

const getRotationMatrix = (degree) => ({
  a: Math.round(Math.cos(degree)),
  b: Math.round(Math.sin(degree)),
  c: -Math.round(Math.sin(degree)),
  d: Math.round(Math.cos(degree)),
  e: 0,
  f: 0,
});

const getTranslationMatrix = (x, y) =>
  ({ a: 1, b: 0, c: 0, d: 1, e: x, f: y });

/**
 * Rotates the given SVG by transforming it's matrix
 * @param {number} rads - radians to rotate the SVG by
 * @param {object} SVG - object containing:
 *  a, b, c, d, e, f attributes describing the SVG transform matrix
 *  SVGWidth, SVGHeight - size of the SVG
 * @returns the given SVG object with adjusted matrix attributes. (rotated SVG)
 */
export const rotate = (rads, SVG) => {
  let m = {a: SVG.a, b: SVG.b, c: SVG.c, d: SVG.d, e: SVG.e, f: SVG.f};
  m = multiply(m, getTranslationMatrix(SVG.SVGWidth/2, SVG.SVGHeight/2));
  m = multiply(m, getRotationMatrix(rads));
  m = multiply(m, getTranslationMatrix(-SVG.SVGWidth/2, -SVG.SVGHeight/2));
  return Object.assign({}, SVG, m);
};
```

In my opinion it would be nice if there were possibility to disable autoPanIfNeeded by props.

Is there a better possibility to rotate svg ?

**Comment by chrvadala** [comment]
Hi farmak. At this moment react svg pan zoom doesn't have a rotate function, but I suggest to work with an inner `<g>` that apply a rotation.
Something like this:
```
<ReactSVGPanZoom>
  <svg....>
     <g transform="translate(....) rotate(...) translate(....)">
         <!--image here -->
    </g>
  </svg>
</ReactSVGPanZoom>
```

If you want to turn off auto pan feature you should set `detectAutoPan={false}`

**Comment by farmasek** [comment]
Thank you for your advice, group rotation will most likely fix my problem, but i wanted to keep  all transformation in your matrix.

I tried changing detectAutoPan but it didn't change anything.

**Comment by chrvadala** [comment]
I have just tested it on the demo http://chrvadala.github.io/react-svg-pan-zoom/ and it seems to work properly.
Could you send a jsfiddle that show this issue?

**Comment by farmasek** [comment]
When i would demonstrate it on your demo page -settings : toolbar position: none, tool:auto.

When you try to drag move svg out of border it wont be allowed, but when you move close to border as possible and then zoom image out it, it is possible and then it behaves unexpectedly. On some click it will return back to rectangle, but not always. If there would be possibility to turn this feature off it would be great.(it does with or without detectAutoPan) It would also fix my boundaries problem when manually calculate matrix values.

**Comment by chrvadala** [comment]
I found which is the problem, but I guess that I should fix this wrong behaviour and not just remove it.
I will start very soon to fix this bug, but I don't know how long it will need.

If you can wait I suggest to fork this project and remove this lines https://github.com/chrvadala/react-svg-pan-zoom/blob/master/src/features/pan.js#L23-L30 (you need to rebuild with `npm run build`)

**Comment by farmasek** [comment]
Great, i am not suggesting to remove it, its great feature, you could  just add possibility to disable it via props.

**Comment by chrvadala** [comment]
I agree. This seems the better way.

**Comment by chrvadala** [comment]
@farmasek 
I have added a new prop `preventPanOutside`

Could you try this new version (not yes released on npm)?
"react-svg-pan-zoom": "chrvadala/react-svg-pan-zoom#f7dc6b92692d5fb9ae09714bc0d530a1ebc3fc45"

**Comment by chrvadala** [comment]
fixed with v2.5
